nat/traversal/pinger.go

```diff
@@ -41,7 +41,7 @@ var (
 
 // NATProviderPinger pings provider and optionally hands off connection to consumer proxy.
 type NATProviderPinger interface {
-	PingProvider(ip string, providerPort, consumerPort, proxyPort int, stop <-chan struct{}) error
+	PingProvider(ip string, cPorts, pPorts []int, proxyPort int) (*net.UDPConn, error)
```

Comment:
Not sure if it's a good idea to remove stop channel. If user cancels connection how will you stop pinger?

Reply:
There is a timeout for pinger, even if it is not stopped it will work for couple of seconds in any case and its conflict with nothing.

Reply:
Default timeout is 10 seconds. If user cancel after 2 seconds there will still be 80 (8 seconds * 10 ports) not needed pings.

Reply:
There is a `func (p *Pinger) Stop()` method which can be called to stop pings.